tests/python_simulators_test/solar_pysam_test.py

```diff
@@ -0,0 +1,89 @@
+"""This module provides unit tests for 'SolarPySAM'."""
+import os
+
+import pytest
+from hercules.python_simulators.solar_pysam import SolarPySAM
+from numpy.testing import assert_almost_equal
+
+
+def get_solar_params():
+
+    full_path = os.path.realpath(__file__)
+    path = os.path.dirname(full_path)
+
+    solar_dict = {
+        "py_sim_type": SolarPySAM,
+        "weather_file_name": None,
+        "weather_data_input": {
+        "Timestamp": ['2018-05-10 12:31:00+00:00'],
+        "SRRL BMS Direct Normal Irradiance (W/m²_irr)": [330.8601989746094],
+        "SRRL BMS Diffuse Horizontal Irradiance (W/m²_irr)": [68.23037719726561],
+        "SRRL BMS Global Horizontal Irradiance (W/m²_irr)": [68.23037719726561],
+        "SRRL BMS Wind Speed at 19' (m/s)": [0.4400002620664621],
+        "SRRL BMS Dry Bulb Temperature (°C)": [11.990000406901045],
+        },
+
+        "system_info_file_name": path+
+            '/../../example_case_folders/07_amr_wind_standin_and_solar_pysam/100MW_1axis_pvsamv1.json',
+        "initial_conditions": {
+            "power": 25, 
+            "irradiance": 1000
+        },
+    }
+    
+    dt = 0.5 # s - input file has a dt of 1 min
+
+    return solar_dict, dt
+
+def create_solar_pysam():
+    solar_dict, dt = get_solar_params()
+    return SolarPySAM(solar_dict, dt)
+
+@pytest.fixture
+def SPS():
+    return create_solar_pysam()
+
+def test_init():
+    solar_dict, dt = get_solar_params()
+    SPS = SolarPySAM(solar_dict, dt)
+
+    assert SPS.dt == dt
+
+    assert SPS.power_mw == solar_dict["initial_conditions"]["power"]
+    assert SPS.dc_power_mw == solar_dict["initial_conditions"]["power"]
+    assert SPS.irradiance == solar_dict["initial_conditions"]["irradiance"]
+    assert SPS.aoi == 0
+
+def test_return_outputs(SPS: SolarPySAM):
+    # outputs after initialization - all outputs should reflect input dict
+    outputs_init = SPS.return_outputs()
+
+    assert outputs_init["power"] == 25
+    assert outputs_init["irradiance"] == 1000
+
+    # change simple battery state as if during simulation
```

Comment:
Can we update the comments so that it's more clear what the part of the solar simulation it's testing?

Reply:
Yes, good idea.